cinema/models.py

```diff
@@ -41,10 +50,73 @@ def __str__(self):
 
 
 class Seat(models.Model):
-
     hall = models.ForeignKey(CinemaHall, on_delete=models.CASCADE, related_name='seats')
     code = models.CharField(max_length=4, default='1A')
     occupied = models.BooleanField(default=False)
 
     def __str__(self):
         return f'Seat {self.code} in hall {str(self.hall)}'
+
+
+class Contact(models.Model):
+    name = models.CharField(max_length=40)
+    email = models.EmailField()
+    phone = PhoneNumberField()
+    city = models.CharField(max_length=20, blank=True, null=True)
+    cinema = models.CharField(max_length=30, blank=True, null=True)
+    subject = models.CharField(max_length=100)
+    message = models.CharField(max_length=1500)
+
+    def __str__(self):
+        return self.email
+
+
+class Schedule(models.Model):
+    movie = models.ForeignKey(Movie, on_delete=models.CASCADE)
+    cinema = models.ForeignKey(Cinema, on_delete=models.CASCADE)
+    hall = ChainedForeignKey(
+        CinemaHall,
+        chained_field='cinema',
+        chained_model_field='cinema',
+        show_all=False,
+        auto_choose=True,
+        sort=True,
+    )
+    playing_time = models.DateTimeField()
+
+    @cached_property
+    def end_time(self):
+        end_time = self.playing_time
+        parsed_time = re.findall(r'\d+[ ]?', self.movie.length)
```

Comment:
The actual movie length field format looks better for me. Also, this is the standard format for IMDb, so if I want to do some scraping at some point, it will be easier for me.